sdk/keyvault/Azure.Security.KeyVault.Keys/src/Cryptography/ByteExtensions.cs

```diff
@@ -0,0 +1,127 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+
+namespace Azure.Security.KeyVault.Keys.Cryptography
+{
+    internal static class ByteExtensions
+    {
+        internal static bool SequenceEqualConstantTime(this byte[] self, byte[] other)
+        {
+            if (self == null)
+                throw new ArgumentNullException(nameof(self));
+
+            if (other == null)
+                throw new ArgumentNullException(nameof(other));
+
+            // Constant time comparison of two byte arrays
+            uint difference = (uint)self.Length ^ (uint)other.Length;
+
+            for (var i = 0; i < self.Length && i < other.Length; i++)
+            {
+                difference |= (uint)(self[i] ^ other[i]);
+            }
+
+            return difference == 0;
+        }
+
+        internal static byte[] Or(this byte[] self, byte[] other)
+        {
+            return Or(self, other, 0);
+        }
+
+        internal static byte[] Or(this byte[] self, byte[] other, int offset)
+        {
+            if (self == null)
+                throw new ArgumentNullException(nameof(self));
+
+            if (other == null)
+                throw new ArgumentNullException(nameof(other));
+
+            if (self.Length > other.Length - offset)
+                throw new ArgumentException("self and other lengths do not match");
+
+            var result = new byte[self.Length];
+
+            for (var i = 0; i < self.Length; i++)
+            {
+                result[i] = (byte)(self[i] | other[offset + i]);
+            }
+
+            return result;
+        }
+
+        internal static byte[] Xor(this byte[] self, byte[] other, bool inPlace = false)
+        {
+            return Xor(self, other, 0, inPlace);
+        }
+
+        internal static byte[] Xor(this byte[] self, byte[] other, int offset, bool inPlace = false)
+        {
+            if (self == null)
+                throw new ArgumentNullException(nameof(self));
+
+            if (other == null)
+                throw new ArgumentNullException(nameof(other));
+
+            if (self.Length > other.Length - offset)
+                throw new ArgumentException("self and other lengths do not match");
+
+            if (inPlace)
```

Comment:
Rather than duplication, just set `result` to a new byte array or `self` based on `inPlace`.